### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by josex2r**
Added tests + Travis CI integration

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "parser": "babel-eslint",
+  "extends": "eslint-config-bbva",
```
Comment (xe21500) [inline .eslintrc:3]: Aren't you missing this package on `package.json`?

**Review by shokmaster — APPROVED** [review]

File: tests/remove-spaces.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const assert = require('assert');
+const expect = require('chai').expect;
+const HTMLBarsMinifier = require('../lib/htmlbars-minifier');
+
+const minifier = new HTMLBarsMinifier('foo', {
+	removeSpacesAroundTags: true
+});
+
+describe('removeSpacesAroundTags', () => {
+	it('should escape special chars', () => {
```
Comment (xe21500) [inline tests/remove-spaces.js:12]: I don't quite understand this test case, you are testing removing spaces, why would you mention special chars now? Also, the test input does not have any space to remove whatsoever.

**Review by adrigzr — APPROVED** [review]

File: tests/remove-spaces.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	it('should remove whitespaces after a DOM element', () => {
+		const input = '<a>    foo';
+
+		assert.equal(
+			minifier.processString(input),
+			'<a>foo'
+		);
+	});
+
+	it('should remove whitespaces between an HTMLBars node and a DOM element', () => {
```
Comment (xe21500) [inline tests/remove-spaces.js]: I would switch the tests here (this one and the next), feels more natural to talk about "HTMLBars node and a DOM element" and then using `{{foo}}     <a>`

File: tests/strip-indentation.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+const expect = require('chai').expect;
+const HTMLBarsMinifier = require('../lib/htmlbars-minifier');
+
+const minifier = new HTMLBarsMinifier('foo', {
+	stripIndentation: true
+});
+
+describe('stripIndentation', () => {
+	describe('from a text node', () => {
+		it('should remove consecutive spaces after a word', () => {
```
Comment (xe21500) [inline tests/strip-indentation.js]: `after` -> `before`

File: tests/strip-indentation.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+			assert.equal(
+				minifier.processString(input),
+				'foo'
+			);
+		});
+	});
+
+	describe('from a DOM element', () => {
+		it('should remove consecutive spaces after a word', () => {
```
Comment (xe21500) [inline tests/strip-indentation.js]: `word` -> `DOM element`, additionally the previous comment about before & after applies here too

File: tests/strip-indentation.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+			assert.equal(
+				minifier.processString(input),
+				'<b>foo</b>'
+			);
+		});
+	});
+
+
+	describe('from a HTMLBars element', () => {
+		it('should remove consecutive spaces after a word', () => {
```
Comment (xe21500) [inline tests/strip-indentation.js]: `word` -> `HTMLBars element` & `after` -> `before`

File: tests/strip-lines.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const assert = require('assert');
+const expect = require('chai').expect;
+const HTMLBarsMinifier = require('../lib/htmlbars-minifier');
+
+const minifier = new HTMLBarsMinifier('foo', {
+	stripIndentation: true,
+	removeTrailingSpaces: true,
```
Comment (xe21500) [inline tests/strip-lines.js:9]: I think you should write these using the default values for everything but the flag you are testing, in this case `stripIndentation`. You already did it for the other tests

File: tests/trailing-spaces.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+			assert.equal(
+				minifier.processString(input),
+				'foo'
+			);
+		});
+	});
+
+	describe('from a DOM element', () => {
+		it('should remove consecutive spaces after a word', () => {
```
Comment (xe21500) [inline tests/trailing-spaces.js]: `word` -> `DOM element`

File: tests/trailing-spaces.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+			assert.equal(
+				minifier.processString(input),
+				'<b>foo</b>'
+			);
+		});
+	});
+
+	describe('from a HTMLBars element', () => {
+		it('should remove consecutive spaces after a word', () => {
```
Comment (xe21500) [inline tests/trailing-spaces.js]: `word` -> `HTMLBars element`

**Review by xe21500 — CHANGES_REQUESTED** [review]

**Review by xe21500 — APPROVED** [review]

File: index.js [hunk 10]
```diff
@@ -1,37 +1,38 @@
...
-        this._super.included.apply(this, arguments);
+		this.options = appOptions['ember-cli-htmlbars-minifier'] || {};
+		for (var option in defaultOptions) {
+			if (!this.options.hasOwnProperty(option)) {
+				this.options[option] = defaultOptions[option];
+			}
+		}
+	},
+	included: function(app) {
+		this._super.included.apply(this, arguments);
```
Comment (beatrizdemiguelperez) [inline index.js:23]: you are initializing config so I think It's better on config hook.  [Docs](https://ember-cli.com/api/classes/Addon.html) said included hook would typically be used to perform additional imports"

File: tests/strip-lines.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+	bar`;
+
+		assert.equal(
+			minifier.processString(input),
+			'{{foo}}bar'
+		);
+	});
+
+	it('should join a block HTMLbars node', () => {
+		const input =
```
Comment (beatrizdemiguelperez) [inline tests/strip-lines.js:38]: indentation here is a little weird

**Pull request merged**